### PR TITLE
Use ERB::Util.url_encode which also escapes '/'

### PR DESCRIPTION
**Pull request by josejulio**
'/' is used on id's and isn't escaped by URI.escape.
 Maybe there is others that aren't handled and is the reason
 `client_utils.rb` was born.

We can't do the escaping on `http_get` unless we pass around the `params`, but we would need to replicate that to `http_post`, `http_put`, etc.

### CODE REVIEW

**Comment by josejulio** [comment]
@cfcosta Is safe to use ERB on the gem?

File: spec/unit/base_spec.rb [hunk 1]
```diff
@@ -255,3 +255,27 @@
...
+    expect(Hawkular::BaseClient.new('not-needed').url('resource/%s', true)).to eql('resource/true')
+  end
+
+  it 'Should work with symbols' do
+    expect(Hawkular::BaseClient.new('not-needed').url('abc/%s', :true)).to eql('abc/true')
+  end
+
+  it 'Should work with multiple params' do
+    expect(Hawkular::BaseClient.new('not-needed').url('resource/%s/inner/%s', true, :ok)).to eql('resource/true/inner/ok')
+  end
```
Comment (jotak) [inline spec/unit/base_spec.rb]: You could add a test to make also sure it passes with "&" and "?" for query params

**Comment by jotak** [comment]
If this encoding ignores "?" and "&", it LGTM

**Comment by josejulio** [comment]
> If this encoding ignores "?" and "&", it LGTM

It doesn't, but it only encodes the params passed. e.g.

`url('resource/%s/stuff?myid=%s&this=that', 'super', '0&delete_everything=true')` will only encode `super` and `0&delete_everything=true`resulting in:
`resource/super/stuff?myid=0%26delete_everything%3Dtrue&this=that`

File: spec/unit/base_spec.rb (hunk 1 above)
Comment (josejulio) [reply spec/unit/base_spec.rb]: Done

**Comment by jotak** [comment]
Oh ok, I see
Looks good to me.

**Review by cfcosta — APPROVED** [review]

**Pull request merged**